cilium-cli/encrypt/ipsec_new_key.go

```
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package encrypt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium/cilium-cli/defaults"
+)
+
+func (s *Encrypt) IPsecNewKey(ctx context.Context) error {
+	ctx, cancelFn := context.WithTimeout(ctx, s.params.WaitDuration)
+	defer cancelFn()
+
+	if err := s.checkEncryptionSecretNotExists(ctx); err != nil {
+		return err
+	}
+
+	newKey, err := createIPsecKey(s.params.IPsecKeyAuthAlgo)
+	if err != nil {
+		return fmt.Errorf("failed to create IPsec key: %w", err)
+	}
+
+	if err := s.createEncryptionSecret(ctx, newKey); err != nil {
+		return err
+	}
+
+	_, err = fmt.Printf("IPsec key successfully created, new key SPI: %d\n", newKey.spi)
```

Do we need to check the error of printf, think CI/unit tests will fail on bad formatting?

---

I don't have a strong opinion here, does it better to drop the error and return nil:
```
	fmt.Printf("IPsec key successfully created, new key SPI: %d\n", newKey.spi)
	return nil
```
?

---

As a general rule it is bad practice to ignore errors.

But of course every rule has an exception :facepalm: ; The Go fmt package upstream documentation states that it is idiomatic to ignore errors specifically from `fmt.Printf()`. https://pkg.go.dev/fmt#Printf

---

Logically here the action (create the key) has actually been performed successfully. If there's some issue with getting this message to stdout then that is not consequential to executing the functionality requested by the user. That suggests that the error from `fmt.Printf()` should _not_ be returned. Again this is something I would typically not recommend but in this specific case it seems like the right pattern (to ignore the error from `fmt.Printf()`).

---

I guess to trace the full answer you'd want to look into the `Write()` implementation for `os.Stdout` https://cs.opensource.google/go/go/+/refs/tags/go1.24.3:src/os/file.go;l=191 ; at a glance the failure conditions are things like if this binary was outputting to stdout and that output was piped to another program which prematurely broke the pipe before reading all the input. I didn't dig much further than that, but I think that gives some reassurance that ignoring the error from fmt.Printf is OK here.

(Incidentally in a traditional unix command you would just silently return a zero exit code to indicate that the command executed successfully; printing like this would not be required. However printing a message like this can be useful if you're wanting to return information back to the user or just provide the user with more confidence that the command was completed successfully).

---

@joestringer thank you for such detailed explanation!
I've removed the error check.